### PR TITLE
HBASE-22509 Address findbugs/spotbugs complaints (branch-1.4)

### PR DESCRIPTION
**Pull request by apurtell**

### CODE REVIEW

File: hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java [hunk 1]
```diff
@@ -7963,7 +7963,9 @@ public Result append(Append mutate, long nonceGroup, long nonce) throws IOExcept
         for (Map.Entry<Store, List<Cell>> entry: removedCellsForMemStore.entrySet()) {
           entry.getKey().add(entry.getValue());
         }
-        if (we != null) mvcc.complete(we);
+        if (we != null) {
```
Comment (apurtell) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java]: Changes to this file are a carrying forward of formatting changes from the branch-1.3 changes proposed on HBASE-22508. They don't address a findbugs issue, only try to keep the code as consistent as possible between branches.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 53 | Docker mode activated. |
||| _ Prechecks _ |
| 0 | findbugs | 0 | Findbugs executables are not available. |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ branch-1.4 Compile Tests _ |
| 0 | mvndep | 15 | Maven dependency ordering for branch |
| +1 | mvninstall | 132 | branch-1.4 passed |
| +1 | compile | 72 | branch-1.4 passed with JDK v1.8.0_212 |
| +1 | compile | 80 | branch-1.4 passed with JDK v1.7.0_222 |
| +1 | checkstyle | 124 | branch-1.4 passed |
| +1 | shadedjars | 164 | branch has no errors when building our shaded downstream artifacts. |
| +1 | javadoc | 62 | branch-1.4 passed with JDK v1.8.0_212 |
| +1 | javadoc | 81 | branch-1.4 passed with JDK v1.7.0_222 |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 15 | Maven dependency ordering for patch |
| +1 | mvninstall | 123 | the patch passed |
| +1 | compile | 76 | the patch passed with JDK v1.8.0_212 |
| +1 | javac | 76 | the patch passed |
| +1 | compile | 84 | the patch passed with JDK v1.7.0_222 |
| +1 | javac | 84 | the patch passed |
| +1 | checkstyle | 26 | The patch passed checkstyle in hbase-common |
| +1 | checkstyle | 12 | The patch passed checkstyle in hbase-hadoop2-compat |
| +1 | checkstyle | 89 | hbase-server: The patch generated 0 new + 368 unchanged - 3 fixed = 368 total (was 371) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 179 | patch has no errors when building our shaded downstream artifacts. |
| -1 | hadoopcheck | 78 | The patch causes 26 errors with Hadoop v2.4.1. |
| -1 | hadoopcheck | 170 | The patch causes 26 errors with Hadoop v2.5.2. |
| -1 | hadoopcheck | 251 | The patch causes 16 errors with Hadoop v2.6.5. |
| +1 | javadoc | 62 | the patch passed with JDK v1.8.0_212 |
| +1 | javadoc | 80 | the patch passed with JDK v1.7.0_222 |
||| _ Other Tests _ |
| +1 | unit | 150 | hbase-common in the patch passed. |
| +1 | unit | 31 | hbase-hadoop2-compat in the patch passed. |
| -1 | unit | 11055 | hbase-server in the patch failed. |
| +1 | asflicense | 65 | The patch does not generate ASF License warnings. |
| | | 13289 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.TestFullLogReconstruction |
|   | hadoop.hbase.master.normalizer.TestSimpleRegionNormalizerOnCluster |
|   | hadoop.hbase.master.procedure.TestDeleteColumnFamilyProcedure |
|   | hadoop.hbase.regionserver.TestRegionReplicaFailover |
|   | hadoop.hbase.master.TestMasterBalanceThrottling |
|   | hadoop.hbase.mapreduce.TestLoadIncrementalHFilesUseSecurityEndPoint |
|   | hadoop.hbase.master.procedure.TestEnableTableProcedure |
|   | hadoop.hbase.regionserver.TestSplitTransactionOnCluster |
|   | hadoop.hbase.master.procedure.TestCreateTableProcedure |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-277/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/277 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux c5095b3da94a 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | branch-1.4 / a4ac174 |
| maven | version: Apache Maven 3.0.5 |
| Default Java | 1.7.0_222 |
| Multi-JDK versions |  /usr/lib/jvm/java-8-openjdk-amd64:1.8.0_212 /usr/lib/jvm/java-7-openjdk-amd64:1.7.0_222 |
| hadoopcheck | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-277/1/artifact/out/patch-javac-2.4.1.txt |
| hadoopcheck | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-277/1/artifact/out/patch-javac-2.5.2.txt |
| hadoopcheck | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-277/1/artifact/out/patch-javac-2.6.5.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-277/1/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-277/1/testReport/ |
| Max. process+thread count | 4069 (vs. ulimit of 10000) |
| modules | C: hbase-common hbase-hadoop2-compat hbase-server U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-277/1/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by apurtell** [comment]
The precommit results looks environmental in nature. Failures begin with ZK session timeouts. This patch is almost identical to the 1.3 and branch-1 patches, both of which had good precommit results. I' still waiting for a valid result on the JIRA (https://issues.apache.org/jira/browse/HBASE-22509). Let me recommit and force push to trigger another run here.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 35 | Docker mode activated. |
||| _ Prechecks _ |
| 0 | findbugs | 0 | Findbugs executables are not available. |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ branch-1.4 Compile Tests _ |
| 0 | mvndep | 20 | Maven dependency ordering for branch |
| +1 | mvninstall | 158 | branch-1.4 passed |
| +1 | compile | 67 | branch-1.4 passed with JDK v1.8.0_212 |
| +1 | compile | 91 | branch-1.4 passed with JDK v1.7.0_222 |
| +1 | checkstyle | 153 | branch-1.4 passed |
| +1 | shadedjars | 150 | branch has no errors when building our shaded downstream artifacts. |
| +1 | javadoc | 59 | branch-1.4 passed with JDK v1.8.0_212 |
| +1 | javadoc | 72 | branch-1.4 passed with JDK v1.7.0_222 |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 12 | Maven dependency ordering for patch |
| +1 | mvninstall | 91 | the patch passed |
| +1 | compile | 64 | the patch passed with JDK v1.8.0_212 |
| +1 | javac | 64 | the patch passed |
| +1 | compile | 75 | the patch passed with JDK v1.7.0_222 |
| +1 | javac | 75 | the patch passed |
| -1 | checkstyle | 84 | hbase-server: The patch generated 3 new + 365 unchanged - 6 fixed = 368 total (was 371) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 152 | patch has no errors when building our shaded downstream artifacts. |
| -1 | hadoopcheck | 84 | The patch causes 26 errors with Hadoop v2.4.1. |
| -1 | hadoopcheck | 169 | The patch causes 26 errors with Hadoop v2.5.2. |
| -1 | hadoopcheck | 263 | The patch causes 16 errors with Hadoop v2.6.5. |
| +1 | javadoc | 56 | the patch passed with JDK v1.8.0_212 |
| +1 | javadoc | 77 | the patch passed with JDK v1.7.0_222 |
||| _ Other Tests _ |
| +1 | unit | 151 | hbase-common in the patch passed. |
| +1 | unit | 32 | hbase-hadoop2-compat in the patch passed. |
| +1 | unit | 6605 | hbase-server in the patch passed. |
| +1 | asflicense | 50 | The patch does not generate ASF License warnings. |
| | | 8750 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-277/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/277 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux fd47202b5fde 4.4.0-139-generic #165-Ubuntu SMP Wed Oct 24 10:58:50 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | branch-1.4 / 8f60ed5 |
| maven | version: Apache Maven 3.0.5 |
| Default Java | 1.7.0_222 |
| Multi-JDK versions |  /usr/lib/jvm/java-8-openjdk-amd64:1.8.0_212 /usr/lib/jvm/java-7-openjdk-amd64:1.7.0_222 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-277/2/artifact/out/diff-checkstyle-hbase-server.txt |
| hadoopcheck | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-277/2/artifact/out/patch-javac-2.4.1.txt |
| hadoopcheck | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-277/2/artifact/out/patch-javac-2.5.2.txt |
| hadoopcheck | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-277/2/artifact/out/patch-javac-2.6.5.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-277/2/testReport/ |
| Max. process+thread count | 4476 (vs. ulimit of 10000) |
| modules | C: hbase-common hbase-hadoop2-compat hbase-server U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-277/2/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by apurtell** [comment]
QA result over on JIRA: https://issues.apache.org/jira/browse/HBASE-22509?focusedCommentId=16853428&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-16853428

hadoopcheck failures are due to another change that introduced test code that twiddles with log levels. Let me follow up separately on that.

Unit test failure here is unrelated to patch:

      TestReplicationKillSlaveRS.killOneSlaveRS ? RetriesExhausted Failed after atte...

The reported checkstyle issues are javadoc nits which are not introduced by this change.

**Comment by busbey** [comment]
those hadoop check versions look wrong. branch-1.4 isn't expected to work with Hadoop < 2.7. Looks like the personality is treating it like branch-1.2 and branch-1.3.

**Comment by busbey** [comment]
filed https://issues.apache.org/jira/browse/HBASE-22518

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 31 | Docker mode activated. |
| -1 | patch | 10 | https://github.com/apache/hbase/pull/277 does not apply to branch-1.4. Rebase required? Wrong Branch? See https://yetus.apache.org/documentation/in-progress/precommit-patchnames for help. |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-277/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/277 |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-277/3/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Pull request merged**